File: android/app/src/main/java/com/anydaf/model/Tractate.kt
package com.anydaf.model

data class Tractate(
    val name: String,
    val startDaf: Int,
    val endDaf: Int,
    /** 0 = amud aleph (a), 1 = amud bet (b). Tamid starts at 25b. */
    val startAmud: Int = 0
) {
    val dafRange: List<Int> get() = (startDaf..endDaf).toList()
}

val allTractates: List<Tractate> = listOf(
    // Seder Zeraim
    Tractate("Berakhot",      2,  64),
    // Seder Moed
    Tractate("Shabbat",       2,  157),
    Tractate("Eiruvin",       2,  105),
    Tractate("Pesachim",      2,  121),
    Tractate("Shekalim",      2,  22),
    Tractate("Rosh Hashanah", 2,  35),
    Tractate("Yoma",          2,  88),
    Tractate("Sukkah",        2,  56),
    Tractate("Beitzah",       2,  40),
    Tractate("Ta\u2019anit",  2,  31),
    Tractate("Megillah",      2,  32),
    Tractate("Moed Katan",    2,  29),
    Tractate("Chagigah",      2,  27),
    // Seder Nashim
    Tractate("Yevamot",       2,  122),
    Tractate("Ketubot",       2,  112),
    Tractate("Nedarim",       2,  91),
    Tractate("Nazir",         2,  66),
    Tractate("Sotah",         2,  49),
    Tractate("Gittin",        2,  90),
    Tractate("Kiddushin",     2,  82),
    // Seder Nezikin
    Tractate("Bava Kamma",    2,  119),
    Tractate("Bava Metzia",   2,  119),
    Tractate("Bava Batra",    2,  176),
    Tractate("Sanhedrin",     2,  113),
    Tractate("Makkot",        2,  24),
    Tractate("Shevuot",       2,  49),
    Tractate("Avodah Zarah",  2,  76),
    Tractate("Horayot",       2,  14),
    // Seder Kodashim
    Tractate("Zevachim",      2,  120),
    Tractate("Menachot",      2,  110),
    Tractate("Hullin",        2,  142),
    Tractate("Bekhorot",      2,  61),
    Tractate("Arakhin",       2,  34),
    Tractate("Temurah",       2,  34),
    Tractate("Keritot",       2,  28),
    Tractate("Meilah",        2,  22),
    Tractate("Kinnim",        22, 25),
    Tractate("Tamid",         25, 33, startAmud = 1),
    Tractate("Middot",        34, 37),
    // Seder Taharot
    Tractate("Niddah",        2,  73),
)

File: AnyDaf/Tractate.swift
import Foundation

struct Tractate: Identifiable, Hashable {
    let id = UUID()
    let name: String
    let startDaf: Int
    let endDaf: Int
    /// 0 = amud aleph (a), 1 = amud bet (b). Tamid starts at 25b.
    let startAmud: Int

    init(name: String, startDaf: Int, endDaf: Int, startAmud: Int = 0) {
        self.name = name
        self.startDaf = startDaf
        self.endDaf = endDaf
        self.startAmud = startAmud
    }

    var dafRange: [Int] {
        Array(startDaf...endDaf)
    }
}

let allTractates: [Tractate] = [
    // Seder Zeraim
    Tractate(name: "Berakhot",      startDaf: 2,  endDaf: 64),
    // Seder Moed
    Tractate(name: "Shabbat",       startDaf: 2,  endDaf: 157),
    Tractate(name: "Eiruvin",       startDaf: 2,  endDaf: 105),
    Tractate(name: "Pesachim",      startDaf: 2,  endDaf: 121),
    Tractate(name: "Shekalim",      startDaf: 2,  endDaf: 22),
    Tractate(name: "Rosh Hashanah", startDaf: 2,  endDaf: 35),
    Tractate(name: "Yoma",          startDaf: 2,  endDaf: 88),
    Tractate(name: "Sukkah",        startDaf: 2,  endDaf: 56),
    Tractate(name: "Beitzah",       startDaf: 2,  endDaf: 40),
    Tractate(name: "Ta\u{2019}anit", startDaf: 2,  endDaf: 31),
    Tractate(name: "Megillah",      startDaf: 2,  endDaf: 32),
    Tractate(name: "Moed Katan",    startDaf: 2,  endDaf: 29),
    Tractate(name: "Chagigah",      startDaf: 2,  endDaf: 27),
    // Seder Nashim
    Tractate(name: "Yevamot",       startDaf: 2,  endDaf: 122),
    Tractate(name: "Ketubot",       startDaf: 2,  endDaf: 112),
    Tractate(name: "Nedarim",       startDaf: 2,  endDaf: 91),
    Tractate(name: "Nazir",         startDaf: 2,  endDaf: 66),
    Tractate(name: "Sotah",         startDaf: 2,  endDaf: 49),
    Tractate(name: "Gittin",        startDaf: 2,  endDaf: 90),
    Tractate(name: "Kiddushin",     startDaf: 2,  endDaf: 82),
    // Seder Nezikin
    Tractate(name: "Bava Kamma",    startDaf: 2,  endDaf: 119),
    Tractate(name: "Bava Metzia",   startDaf: 2,  endDaf: 119),
    Tractate(name: "Bava Batra",    startDaf: 2,  endDaf: 176),
    Tractate(name: "Sanhedrin",     startDaf: 2,  endDaf: 113),
    Tractate(name: "Makkot",        startDaf: 2,  endDaf: 24),
    Tractate(name: "Shevuot",       startDaf: 2,  endDaf: 49),
    Tractate(name: "Avodah Zarah",  startDaf: 2,  endDaf: 76),
    Tractate(name: "Horayot",       startDaf: 2,  endDaf: 14),
    // Seder Kodashim
    Tractate(name: "Zevachim",     startDaf: 2,  endDaf: 120),
    Tractate(name: "Menachot",      startDaf: 2,  endDaf: 110),
    Tractate(name: "Hullin",        startDaf: 2,  endDaf: 142),
    Tractate(name: "Bekhorot",      startDaf: 2,  endDaf: 61),
    Tractate(name: "Arakhin",       startDaf: 2,  endDaf: 34),
    Tractate(name: "Temurah",       startDaf: 2,  endDaf: 34),
    Tractate(name: "Keritot",       startDaf: 2,  endDaf: 28),
    Tractate(name: "Meilah",        startDaf: 2,  endDaf: 22),
    Tractate(name: "Kinnim",        startDaf: 22, endDaf: 25),
    Tractate(name: "Tamid",         startDaf: 25, endDaf: 33, startAmud: 1),
    Tractate(name: "Middot",        startDaf: 34, endDaf: 37),
    // Seder Taharot
    Tractate(name: "Niddah",        startDaf: 2,  endDaf: 73),
]
